### PR TITLE
Will 1items merchant

### PR DESCRIPTION
**Pull request by williamfchen**
#### Updates Made -
- User Stories Updated: US12 added, 5 most popular items per merchant index
- Database/Migrations: 
- Routes:
- Models: item class method added
- Controllers:
- Views: merchant item index displays most popular items
- Testing: Added factorybot factories and config.  Utilized faker in factory creation.
  - Models: US12 test
  - Features:
- Considerations(Extra notes, refactors, etc.):

### CODE REVIEW

File: app/views/merchants/items/index.html.erb [hunk 1]
```diff
@@ -1,4 +1,29 @@
...
+        <td><%= link_to "#{item.name}", merchant_item_path(@merchant, item), method: :get %></td>
+        <!- <td><%= item.description %></td> ->
+        <!- <td><%= number_to_currency(item.invoice_items.successful_average_item_price / 100) %></td> ->
+        <!- <td><%= item.invoice_items.successful_total_item_quantity %></td> ->
+        <td><%= number_to_currency(item.revenue / 100.0) %></td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+
```
Comment (pcbennett108) [inline app/views/merchants/items/index.html.erb:21]: I'm curious about the   <!-     ->   tags. A comment has 2 dashes, but I've never come across the single dash.

File: app/models/item.rb [hunk 2]
```diff
@@ -15,4 +15,29 @@ def formatted_unit_price
...
+    Item
+    .joins(:invoice_items)
+    .joins(invoices: :transactions)
+    .select('items.*, sum(invoice_items.quantity * invoice_items.unit_price) as revenue')
+    .where('transactions.result = ?', 1)
+    .group(:id)
+    .order('revenue DESC')
+    .limit(5)
+  end
+
```
Comment (kbug819) [inline app/models/item.rb:29]: This looks great!

File: spec/factories/transactions.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :transaction do
+    invoice
+    credit_card_number { Faker::Finance.credit_card }
+    credit_card_expiration_date { Faker::Business.credit_card_expiry_date }
+    result { [0, 1].sample }
+  end
+end
```
Comment (kbug819) [inline spec/factories/transactions.rb:8]: Thanks so much for setting up Factory Bot!

Comment (pcbennett108) [reply spec/factories/transactions.rb:8]: loving the factory bot

**Review by pcbennett108 — APPROVED** [review]
fantastic!

File: app/models/item.rb [hunk 4]
```diff
@@ -13,4 +13,29 @@ def formatted_unit_price
...
```
Comment (Antoine-Aube) [inline app/models/item.rb:29]: Agreed! Solid query. DIdn't know you could make a calculation within an aggregation! super cool.

**Pull request merged**